dart/common/NameManager.h

```diff
@@ -189,6 +227,12 @@ class NameManager
     return (mMap.find(_name) != mMap.end());
   }
 
+  /// Return true if the object is contained
+  bool hasObject(T _obj) const
+  {
+    return (mReverseMap.find(_obj) != mReverseMap.end());
+  }
+
   /// Get the number of the objects currently stored by the NameManager
   size_t getCount() const
   {
```

Comment:
If it causes noticeable overhead then we might need to find other way, but `assert(mMap.size() == mReverseMap.size())` wouldn't be the case I think. But still it might not be able to capture the violation if `getCount()` is overloaded by the inherited class.

It seems fine without a debug mode check, I thought it would save the time when it causes problem.